op-node/p2p/store/records_book.go

```diff
@@ -80,26 +84,24 @@ func (d *recordsBook[K, V]) startGC() {
 	startGc(d.ctx, d.log, d.clock, &d.bgTasks, d.prune)
 }
 
-func (d *recordsBook[K, V]) GetRecord(key K) (V, error) {
-	d.RLock()
-	defer d.RUnlock()
-	rec, err := d.getRecord(key)
-	return rec, err
-}
-
 func (d *recordsBook[K, V]) dsKey(key K) ds.Key {
 	return d.dsBaseKey.Child(d.dsEntryKey(key))
 }
 
 func (d *recordsBook[K, V]) deleteRecord(key K) error {
-	d.cache.Remove(key)
+	// If access to this isn't synchronized, removing from the cache first can result in the stored
+	// item being cached again before it is deleted.
 	err := d.store.Delete(d.ctx, d.dsKey(key))
+	d.cache.Remove(key)
 	if err == nil || errors.Is(err, ds.ErrNotFound) {
 		return nil
 	}
 	return fmt.Errorf("failed to delete entry with key %v: %w", key, err)
 }
 
```

Comment:
Smal nit on godoc style - it usually starts with the name of the function as the first word in the description with an explanation of what the function does and anything extra somebody calling the function should know. This is not done perfectly throughout the codebase right now

Reply:
I think it was a mistake to recommend that style in Go. See https://groups.google.com/g/golang-nuts/c/_SuyQsr6tuA/m/rA-d07S9oh0J for some of the original benefits. I usually avoid naming the thing at all if I can. I can add it if you think it's better.

Reply:
We still follow that convention generally, but not perfectly, so I'd try in general to follow it. Since this is an internal function, it's not super important so I'm fine with either. We can have an internal discussion on whether we want to continue following it. At least for exported functions I think it's necessary to generate correct godoc and linters fill fail otherwise.